src/exporters/cds.py

```diff
@@ -0,0 +1,66 @@
+import cdsapi
+from pathlib import Path
+import warnings
+
+from typing import Dict
+
+from .base import BaseExporter
+
+
+class CDSExporter(BaseExporter):
+    """Exports for the Climate Data Store
+
+    cds.climate.copernicus.eu
+    """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+        super().__init__(data_folder)
+        self.client = cdsapi.Client()
+
+    @staticmethod
+    def make_filename(dataset: str, selection_request: Dict) -> str:
+        """Makes the appropriate filename for a CDS export
+        """
+        date_str = ''
+        if 'year' in selection_request:
+            years = selection_request['year']
+            if len(years) > 1:
+                warnings.warn('More than 1 year of data being exported! '
+                              'Export times may be significant.')
+                years.sort()
+                date_str = f'{years[0]}_{years[-1]}'
+            else:
+                date_str = str(years[0])
+        elif 'date' in selection_request:
+            date_str = selection_request['date'].replace('/', '_')
+
+        # force all data exports to be in netcdf format
+        # TODO: This is not possible for some exports. We should select
+        # our preferences and force those choices
+        selection_request['format'] = 'netcdf'
```

Comment:
TODO: transform non-netcdf files into netcdf files as part of the export

Reply:
we can use that `engine='cfgrib'` argument using `ds = xarray.open_dataset(filepath, engine='cfgrib')` as [here](https://github.com/tommylees112/esowc_notes/blob/master/notebooks/01_gt_exploration.ipynb) 

and then you can save using:
`ds.to_netcdf(output_filename)`